### PR TITLE
Changes to music list

### PR DESCRIPTION
**Pull request by likeawindrammer**

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -1349,10 +1351,11 @@ void Courtroom::list_music()
     ++n_listed_songs;
   }
 
-  ui_music_list->expandAll(); // Needs to somehow remember which categories were
-                              // expanded/collapsed if the music list didn't
-                              // change since last time
+  ui_music_list->expandAll();
   if (ui_music_search->text() != "") {
+      ui_music_list->expandAll(); // Needs to somehow remember which categories were
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:1356]: Why is expandAll() called twice? Wouldn't this if clause not matter at all?

File: src/courtroom.cpp [hunk 2]
```diff
@@ -3510,6 +3513,7 @@ void Courtroom::on_ooc_toggle_clicked()
 // Todo: multithread this due to some servers having large as hell music list
 void Courtroom::on_music_search_edited(QString p_text)
 {
+    ui_music_list->expandAll();
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:3516]: Music list will be expanded every time you add anything to the search string regardless of your input as a user... Is this even good ui design???

File: src/courtroom.cpp [hunk 3]
```diff
@@ -3552,6 +3556,13 @@ void Courtroom::on_music_search_edited(QString p_text)
   }
 }
 
+void Courtroom::on_music_search_return_pressed()
+{
+  if (ui_music_search->text() == "") {
+    ui_music_list->collapseAll();
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:3562]: This is kind of a cool alternative to right click -> expand all, me like.

**Comment by Crystalwarrior** [comment]
oops I shoulda made my review "needs to be resovled" oh well, still new to this review thing

File: src/courtroom.cpp (hunk 3 above)
Comment (oldmud0) [reply src/courtroom.cpp:3562]: Typically expand all can also be done with the numpad asterisk key.

File: src/courtroom.cpp [hunk 4]
```diff
@@ -322,6 +322,8 @@ Courtroom::Courtroom(AOApplication *p_ao_app) : QMainWindow()
   connect(ui_effects_dropdown, SIGNAL(customContextMenuRequested(QPoint)), this,
           SLOT(on_effects_context_menu_requested(QPoint)));
 
+  connect(ui_music_search, SIGNAL(returnPressed()), this,
+          SLOT(on_music_search_keypr()));
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:326]: on_music_search_keypr should be renamed to on_music_search_return_pressed like before, and its signal needs to be added to header

**Pull request merged**